Source/DafnyCore/Auditor/Assumption.cs

```diff
@@ -0,0 +1,79 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+
+namespace Microsoft.Dafny.Auditor;
+
+public record AssumptionDescription(string issue, string mitigation, bool isExplicit) {
+  public static AssumptionDescription AxiomAttributeAssumption = new(
+    issue: "Declaration has explicit [{:axiom}] attribute.",
+    mitigation: "Provide a proof or test.",
+    isExplicit: true);
+  public static AssumptionDescription VerifyFalseAssumption = new(
+    issue: "Declaration has [{:verify false}] attribute.",
+    mitigation: "Remove and prove if possible.",
+    isExplicit: false);
+  public static AssumptionDescription ExternWithRequires = new(
+    issue: "Declaration with [{:extern}] has a requires clause.",
+    mitigation: "Test external caller (maybe with [/testContracts]).",
+    isExplicit: false);
+  public static AssumptionDescription ExternWithEnsures = new(
+    issue: "Declaration with [{:extern}] has a ensures clause.",
+    mitigation: "Test external callee (maybe with [/testContracts]).",
+    isExplicit: false);
+  public static AssumptionDescription AssumeInBody = new(
+    issue: "Definition has [assume] statement in body.",
+    mitigation: "Replace with [assert] and prove or add [{:axiom}].",
+    isExplicit: false);
+  public static AssumptionDescription DecreasesStar = new(
+    issue: "Method may not terminate (uses [decreases *]).",
+    mitigation: "Provide a valid [decreases] clause.",
+    isExplicit: false);
+  public static AssumptionDescription TerminationFalse = new(
+    issue: "Trait method calls may not terminate (uses [{:termination false}]).",
+    mitigation: "Remove if possible.",
+    isExplicit: false);
+
+  public static AssumptionDescription NoBody(bool isGhost) {
+    return new(
+      issue: (isGhost ? "Ghost" : "Compiled") +
+             " declaration has no body and has an ensures clause.",
+      mitigation: "Provide a body or add [{:axiom}].",
+      isExplicit: false);
+  }
+}
+
+public record Assumption(Declaration decl, IEnumerable<AssumptionDescription> assumptions) {
+  public IEnumerable<string> Warnings() {
```

Comment:
Warnings is an existing concept, a message with a source location that shows up in the IDE and in the CLI when using `dafny verify`, and this doesn't seem to refer to those warnings. Can you use a different term?

Reply:
Re-reading this, I think it makes sense to leave it as-is. The purpose of this method is to produce warnings of exactly the type you mention: they'll be displayed in yellow at the CLI and would show up as warnings in the IDE if it ever invoked the auditor code. See [here](https://github.com/dafny-lang/dafny/pull/3175/files#diff-40edc58c676412b086903a0ed18fe28bfaeef19dc7d758876683ac8454dc5139R79-R81).